src/Generator/Passes/ParamTypeToInterfacePass.cs

```diff
@@ -7,20 +7,22 @@ public class ParamTypeToInterfacePass : TranslationUnitPass
     {
         public override bool VisitFunctionDecl(Function function)
         {
-            if (!function.IsOperator)
-            {
+            if (!function.IsOperator || function.Parameters.Count > 1)
```

Comment:
The unary operators must return the containing type while Binary operators can return other types.
Without this check Build was failing in CSharpTemp.cs at operator ++ in Bar.